components/constellation/constellation.rs

```diff
@@ -1615,34 +1641,12 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
             .map(|pipeline_id| self.get_top_level_frame_for_pipeline(pipeline_id))
             .unwrap_or(self.root_frame_id);
 
-        let mut traversal_info = HashMap::new();
-
-        match direction {
-            TraversalDirection::Forward(delta) => {
-                let mut future = self.joint_session_future(top_level_frame_id);
-                for _ in 0..delta {
-                    match future.pop() {
-                        Some((_, frame_id, pipeline_id)) => {
-                            traversal_info.insert(frame_id, pipeline_id);
-                        },
-                        None => return warn!("invalid traversal delta"),
-                    }
-                }
-            },
-            TraversalDirection::Back(delta) => {
-                let mut past = self.joint_session_past(top_level_frame_id);
-                for _ in 0..delta {
-                    match past.pop() {
-                        Some((_, frame_id, pipeline_id)) => {
-                            traversal_info.insert(frame_id, pipeline_id);
-                        },
-                        None => return warn!("invalid traversal delta"),
-                    }
-                }
-            },
+        let traversal_info: HashMap<FrameId, Instant> = match direction {
```

Comment:
This will no longer return earlier in the cases where delta is greater than the jsh future/past